checker/check_test.go

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	. "github.com/onsi/gomega"
+	"testing"
+
+	authz "github.com/envoyproxy/data-plane-api/api/auth"
+	"github.com/projectcalico/app-policy/policystore"
+	"github.com/projectcalico/app-policy/proto"
+)
+
+// ActionFromString should parse strings in case insensitive mode.
+func TestActionFromString(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	g.Expect(ActionFromString("allow")).To(Equal(ALLOW))
+	g.Expect(ActionFromString("Allow")).To(Equal(ALLOW))
+	g.Expect(ActionFromString("deny")).To(Equal(DENY))
+	g.Expect(ActionFromString("Deny")).To(Equal(DENY))
+	g.Expect(ActionFromString("pass")).To(Equal(PASS))
+	g.Expect(ActionFromString("Pass")).To(Equal(PASS))
+	g.Expect(ActionFromString("log")).To(Equal(LOG))
+	g.Expect(ActionFromString("Log")).To(Equal(LOG))
+	g.Expect(func() { ActionFromString("no_match") }).To(Panic())
```

Comment:
A better thing to do (more inline with Go's error handling style) is to return `Action` and `error` from that function, so when there's no match, it returns an error value. Go's error handling style is bubble up the error to the caller and handle it there, so in this case `ActionFromString` could return `(Action, error)` and `checkRules` can log and handle it. It should also make this test assertion easier

Reply:
The caller should not be calling `ActionFromString` with an invalid action; that's the contract for this function. In this design the action strings come from the Policy Sync API and we trust it not to give us invalid data.

If I did want to allow the caller to handle bad values, then I agree with your design. But, I don't.